controllers/notes.js

```diff
@@ -0,0 +1,99 @@
+const note = require('../models/note');
+const { StatusCodes } = require('http-status-codes');
+const parseVErr = require('../utils/parseValidationErr');
+
+const getAllNotes = async (req, res) => {
+  const notes = await note.find({ createdBy: req.user.id }).sort('CreatedAt');
+  res.status(StatusCodes.OK).render('notes', { notes });
+};
+
+const addNote = async (req, res) => {
```

Comment:
In the addNote method, you need to validate if the title, subject, and content are not empty, and if they are empty, return an error using the req.flash method and then redirect to the 'notes/new' route. You can redirect to the "/notes" route if the notes were successfully added. You don't need to find the notes and render the "notes" page because it is already done in the getAllNotes method, which will be called when you redirect to the "/notes" route.

```
try {
    const {title, subject, content} = req.body;

    if (!title || !subject || !content) {
      req.flash("error", "Title, subject or content fields cannot be empty.");
      return res.redirect('/notes/new');
    } 

    req.body.createdBy = req.user.id;
    await noteModel.create(req.body);
    req.flash("info", "The note has been successfully added.")
  } catch (err) {
    req.flash("error", "Undefined error.")
  }
  res.redirect('/notes');
```